expo/expo
# [expo-app-loading][expo] Extract AppLoading to separate module

# Why

`AppLoading` is the last module in Expo that can be extracted into the separate module. Although `expo-splash-screen` is now our source of truth about splash-related things, to keep backward compatibility we need this package as well.
Additionally, we want to remove `expo-splash-screen` dependency from the `expo` module, because it's being auto-installed in bare workflow when the user is using `expo` package and we don't want that (it has unwanted side-effects that are sometimes unwanted).

# How

- created a separate `expo-app-loading` module in `packages` and moved `AppLoading` code there. This packages is a single-react-component package now.
- removed `RootErrorBoundary` as it's now no-op component on every platform (previously it was only affecting the Android platform)
- tweaked & sorted tests and `removed/deprecated` packages lists
- corrected all imports of `{ AppLoading }` or replaced it with `SplashScreen` if possible

# Test Plan

- `expo` tests are passing
- managed `ncl` & `bare-expo` apps are working
- when using the `AppLoading` according to the current example ([example](https://docs.expo.io/versions/latest/sdk/app-loading/#props) `import { AppLoading } from 'expo'`) the developer will encounter the well-known error message in the console:
```
Error: The following APIs have moved to separate packages and importing them from the "expo" package is no longer supported: AppLoading.

1. Add correct versions of these packages to your project using:

   expo install expo-app-loading

   If "install" is not recognized as an expo command, update your expo-cli installation.

2. Change your imports so they use specific packages instead of the "expo" package:

 - import { AppLoading } from 'expo' -> import AppLoading from 'expo-app-loading'
```
- after changing the import according to the suggestion everything works as previously 🎉
